test/ARCHITECTURE.md

```diff
@@ -0,0 +1,263 @@
+# Architecture
+
+This document describes the architecture of Cockpit's browser integration
+tests. The tests should automate how a normal user interacts with Cockpit.
+This requires a test machine which can add multiple disks or interfaces,
+reboot, interact with multiple machines on the same network and run potentially
+destructive test scenarios (e.g. installing/updating packages, formatting
+disks).
+
+For these reasons, Cockpit tests run inside a virtual machine (VM). The virtual
+machine uses Cockpit specific virtual machine images maintained and created in
+the [bots](https://github.com/cockpit-project/bots) repository. The images are
+usually based on a distribution's cloud image customized with:
+
+* A well-known password and SSH key for the admin and root users
+* Packages required to test Cockpit
+* A build chroot with Cockpit's build dependencies to build the to be tested
+  Cockpit source inside the virtual machine offline. This allows a developer on
```

Comment:
The sentence "A build chroot with Cockpit's build dependencies to build the to be tested
  Cockpit source inside the virtual machine offline." sounds confusing.

Reply:
Ok, after reading it a dozen times, it kind of makes sense now :)